insights.go

```diff
@@ -25,11 +31,25 @@ func insightsIsRegistered() (bool, error) {
 	// The `.registered` file is always present on a registered system.
 	err := exec.Command("/usr/bin/insights-client", "--status").Run()
 	if err != nil {
+		// When the error is ExitError, then we know that insights-client only returned
+		// some error code not equal to zero. We do not care about error number.
+		var exitError *exec.ExitError
+		if errors.As(err, &exitError) {
+			// Check if .unregistered file exists. In that case insights-client
+			// is probably in non-legacy mode, and it is only unregistered.
+			// There is hopefully no reason to return any error in such case.
+			_, err := os.Stat("/etc/insights-client/.unregistered")
+			if err == nil {
+				return false, nil
+			}
+		} else {
+			return false, err
+		}
```

Comment:
Why do we care about the exit code from running insights-client? Would it be easier to check for the presence of both `/etc/insights-client/.registered` and `/etc/insights-client/.unregistered` and determine status from the existence or non-existence of those files?

Reply:
Because current `insights-client` is "sub-optimal". You cannot trust to any of sign 100%. You have to combine several signs together to get something at least little bit reliable.

Reply:
You aren't helping yourself by disliking insights-client. You can't trust _any_ software 100%. Just accept its quirkiness as a fact and move on. Some parts of the whole stack are much less reliable than the client itself.

Reply:
It's not about liking or disliking insights-client. I'm only describing current situation. If we relied only on existence of `/etc/insights-client/.registered` and `/etc/insights-client/.unregistered`, then it would not be reliable as it was described in the case in https://github.com/RedHatInsights/rhc/pull/125#discussion_r1747070445.